src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate, String validationRegex, String errorMessage) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        if (!isValidContact(trimmedContact, validationRegex)) {
+            throw new IllegalValueException(errorMessage);
+        }
+        this.value = trimmedContact;
+    }
+ 
+    public String toString() {
+        return value;
+    }
+
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
This works but do remember the case where a contacts of different classes have the same value.